dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -140,9 +140,10 @@ private String buildTrackedEntityInstanceHql( TrackedEntityInstanceQueryParams p
             hql += hlp.whereAnd() + "tei.lastUpdated < '" + getMediumDateString( params.getLastUpdatedEndDate() ) + "'";
         }
 
+        params.handleOrganisationUnits();

```

Comment:
This PR is mixing two issues, could we split it up? Tests seem to be failing @stian-sandvold .